tslearn/neighbors.py

```diff
@@ -12,16 +12,75 @@
 from sklearn.utils.validation import check_is_fitted
 from scipy.spatial.distance import cdist as scipy_cdist
 
-from tslearn.metrics import cdist_dtw, cdist_soft_dtw, VARIABLE_LENGTH_METRICS
+from tslearn.metrics import cdist_dtw, cdist_soft_dtw, \
+    cdist_sax, VARIABLE_LENGTH_METRICS
+from tslearn.piecewise import SymbolicAggregateApproximation
 from tslearn.utils import (to_time_series_dataset, to_sklearn_dataset,
                            check_dims)
 
-neighbors.VALID_METRICS['brute'].extend(['dtw', 'softdtw'])
+neighbors.VALID_METRICS['brute'].extend(['dtw', 'softdtw', 'sax'])
 
 
 class KNeighborsTimeSeriesMixin(KNeighborsMixin):
     """Mixin for k-neighbors searches on Time Series."""
 
+    def _sax_preprocess(self, X, n_segments=10, alphabet_size_avg=4):
+        # First, we standard-normalize the data using mu and sigma
```

Comment:
I don't know. I think this again depends on the type of application... 
![Screenshot from 2020-03-08 10-18-11](https://user-images.githubusercontent.com/5750603/76159968-2ffd4380-6126-11ea-93b7-8f063af7f482.png)

Like here, would you prefer that ts2 is more similar to ts3 than to ts1, or do you prefer that all pairs of timeseries have the same similarity?

Reply:
I would expect all the pairs to have the same similarity, but it's just my point of view. I'm not an expert on this but I don't think that I read a single paper in which they compute the mean and the standard deviation on the whole dataset.

In the SAX paper, they say that they normalize the time series so that each time series has zero mean and unit variance.
<img width="527" alt="Capture d’écran 2020-03-08 à 13 58 18" src="https://user-images.githubusercontent.com/17053934/76163247-d8ba9b80-6144-11ea-93fa-1f461e7d8d10.png">

Reply:
Ok, I missed that part! I'll change it to have time-series-wise normalization later!

Reply:
So I removed all normalization. It is now up to the user to apply timeseries-normalization before running KNN!

Reply:
OK for me. Could you at least document it in the class definition (something like "SAX is supposed to operate on centered-reduced time series, the use of `TimeSeriesScalerMeanVariance` is strongly encouraged")?

Reply:
Ok, I indeed forgot to re-check if docs are still up-to-date. Let me make that change & check up on all docs in a bit!

Reply:
Or maybe in the class definition + the distance function?